plugins/inputs/ctrlx_datalayer/ctrlx_datalayer.go

```diff
@@ -0,0 +1,363 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package ctrlx_datalayer
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/boschrexroth/ctrlx-datalayer-golang/pkg/sseclient"
+	"github.com/boschrexroth/ctrlx-datalayer-golang/pkg/token"
+	"github.com/google/uuid"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/influxdata/telegraf/metric"
+	httpconfig "github.com/influxdata/telegraf/plugins/common/http"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	jsonParser "github.com/influxdata/telegraf/plugins/parsers/json"
+)
+
+// This plugin is based on the official ctrlX CORE API. Documentation can be found in OpenAPI format at:
+// https://boschrexroth.github.io/rest-api-description/ctrlx-automation/ctrlx-core/
+// Used APIs are:
+// * ctrlX CORE - Authorization and Authentication API
+// * ctrlX CORE - Data Layer API
+//
+// All communication between the device and this input plugin is based
+// on https REST and HTML5 Server Sent Events (sse).
+
+//go:embed sample.conf
+var sampleConfig string
+
+// CtrlXDataLayer encapsulated the configuration as well as the state of this plugin.
+type CtrlXDataLayer struct {
+	Hostname string `toml:"hostname"`
```

Comment:
Can this have a different option name like `endpoint` or `server` or something as hostname usually refers to the _current_ machine's name!?

Reply:
We changed the name from `hostname` to `server` for this option, see https://github.com/influxdata/telegraf/pull/11155/commits/7f6b2608a545d7d2695a3b962306da7aead6e1a1